### ISSUE TITLE
Clearwater sprout upgrade process needs a way to equalize memcached data to avoid an outage

### ISSUE THREAD

**Issue by tcaterisano** [Metaswitch/sprout#852]
My understanding is that there is no way for customers to know which sprout nodes they can safely remove during a sprout node upgrade cycle. Another related issue is that some customers have very long re-reg times configured which impacts the planned upgrade process since it relies on subscriber re-reg to populate memcached data in the new sprout nodes. A means of replicating this data across all nodes, at least during an upgrade, would prevent an outage due to this issue. Customers could then take down up to 50% of their nodes during an upgrade and not affect subscribers.

**Comment by mirw** [comment]
@bossmc, does it make sense for you to claim this?

**Comment by bossmc** [comment]
@tcaterisano Rob's made a fix that means that it's ok to resize the deployment by adding or removing any node (not just the one at the end of the `server` list).  The work to pro-actively resynchronise data during scale-up/scale-down is way bigger than a single bug fix so we've scheduled an epic of stories to implement that function over the next few releases.

Closing this issue as it's superseded by the epic.
